crates/bevy_reflect/src/impls/std.rs

```diff
@@ -131,11 +130,6 @@ impl_reflect_value!(NonZeroU16(Debug, Hash, PartialEq, Serialize, Deserialize));
 impl_reflect_value!(NonZeroU8(Debug, Hash, PartialEq, Serialize, Deserialize));
 impl_reflect_value!(NonZeroI8(Debug, Hash, PartialEq, Serialize, Deserialize));
 
-// Only for platforms that can serialize it as in serde:
-// https://github.com/serde-rs/serde/blob/3ffb86fc70efd3d329519e2dddfa306cc04f167c/serde/src/de/impls.rs#L1732
-#[cfg(any(unix, windows))]
-impl_reflect_value!(OsString(Debug, Hash, PartialEq, Serialize, Deserialize));
```

Comment:
Rather than removing these type data registrations, can we just add another impl using the inverse flag of `#[cfg(not(any(unix, windows))]` (or whatever)?

Reply:
I thought that it could confuse users why some things works on one platform and doesn't work on another.
But I could add it if you think that it could be useful.

Reply:
Yeah that makes sense. I think it's worth keeping. Users will just need to be aware of the differences if they intend on supporting other platforms

Reply:
Okay, done!